### ISSUE TITLE
Random read/write performance really low

### ISSUE THREAD

**Issue by christoph-buente** [copperhead/bugtracker#226]
Hi,

i installed Copperhead OS on my LG nexus 5X 32GB. Runs really nice, thank you for the awesome work. But i had a few issues when trying to select a different ringtone. The phone would stall and a popup appears, asking me if i wanted to wait on the app. That got me a bit suspicious. And today i downloaded the latest update for 5X and when the download started the phone would go very slow and also block. I installed AndroBench and checked the read/write performance of the storage. And this are the results:

SEQ RD: 240MB/s
SEQ WR: 156MB/s
RND RD: 6.7MB/s
RND WR: 3.2MB/s

I looked up those results for comparison on the internet and found people reporting for
RND RD: 25MB/s
RND WR: 11.7MB/s

SEQ roughly shows the same results as my phone. Now i'm asking myself is it due to strong encryption in the phone or the security features that drop the performance. Or is it just my phone? I think it's not, because sequention performance seems to be good. And maybe the ringtone issue is related to something else ;(

Any ideas?

Thx, Christoph

**Comment by thestinger** [comment]
The Nexus 5 doesn't use hardware encryption acceleration, unlike the newer devices. CopperheadOS uses the same encryption implementation as Android. Selecting a different ringtone works fine here, and no one else seems to have run into that problem. It seems unlikely that it's a CopperheadOS issue.

**Comment by christoph-buente** [comment]
It's actually the LG nexus 5X, it was a typo in my description. Is there a way to backup the full device, apps, settings? I would flash the vendor image again and check if the problem persists with a google android 6.0.1

**Comment by thestinger** [comment]
If it's a 5X, encryption is enabled by default in stock Android too. There's no difference. You can back up using `adb backup` with the appropriate switches and then `adb restore` after reinstalling.

**Comment by christoph-buente** [comment]
After installing the latest MMB29V.2016.03.27.04.15.10	 for Nexus X5 bullhead and killing all running apps before, the performance is ok now

SEQ RD: 249MB/s
SEQ WR: 157MB/s
RND RD: 23.2MB/s
RND WR: 13.2MB/s

The problem with selecting ringtone and notification sounds persists. I can choose from a very long list, but every other file takes 10+ seconds to load. Sometimes i see the watchdog dialog asking me if i want to wait or kill the app.

**Comment by thestinger** [comment]
Can you open a new issue for that?

**Comment by christoph-buente** [comment]
The latest update seemed to have fixed it somehow. Selection of ringtones works just fine now.
